### DISCUSSION TITLE
enable http2 support by default in pekko-http 1.1.0?

### DISCUSSION THREAD

**Discussion by pjfanning** [apache/pekko-http#438] [Ideas]
currently disabled by default - see https://github.com/apache/incubator-pekko-http/blob/e11260fa02120247016fe385d1e898677e4f3432/http-core/src/main/resources/reference.conf#L31

**Comment by jrudolph** [comment]
It's not "preview" or "experimental" any more, but enabling it changes the runtime behavior significantly and people might not want to accidentally use it in usual deployments.

There's a potential performance component to it, because HTTP2 support needs to hook into the connection to figure out when HTTP2 is requested. Especially, in deployments without TLS, enabling HTTP2 currently automatically enables a somewhat unusual configuration of running HTTP2 on plain connections.

When you want to run HTTP2, you will usually also have to setup TLS manually, so there's little cost to letting it disabled by default in my opinion.

**Reply by He-Pin** [reply]
I'm not sure about this ,but I saw quarkus are optimizating http/2 supporting, so I would like it to be disabled by default, end users can turn on if they need it. 

https://github.com/quarkusio/quarkus/issues/34473
